scripts/veloxbench/veloxbench/cpp_micro_benchmarks.py

```diff
@@ -131,6 +163,58 @@ def _format_unit(x):
         return x
 
 
-if __name__ == "__main__":
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description="VeloxBench Client Tool",
+        epilog="(c) Meta Platforms 2004-present",
+    )
+    parser.add_argument(
+        "--binary_path",
+        default=None,
+        help="Directory where benchmark binaries are stored. "
+        "Defaults to release build directory.",
+    )
+    parser.add_argument(
+        "--binary_filter",
+        default=None,
+        help="Filter applied to binary names. "
+        "By default execute all binaries found.",
+    )
+    parser.add_argument(
+        "--bm_filter",
+        default=None,
+        help="Filter applied to benchmark names within binaries. "
+        "By default execute all benchmarks.",
+    )
+    parser.add_argument(
+        "--bm_max_secs",
+        default=10,
+        type=int,
+        help="For how many second to run each benchmark in a binary.",
+    )
+    parser.add_argument(
+        "--bm_max_trials",
+        default=1000000,
+        type=int,
+        help="Maximum number of trials (iterations) executed for each benchmark.",
+    )
+    return parser.parse_args()
+
+
+def main():
+    args = parse_arguments()
+
     with tempfile.TemporaryDirectory() as result_dir:
-        LocalCppMicroBenchmarks().run(result_dir=result_dir)
```

Comment:
result_dir=result_dir? different variables? I am not super nerd in python but maybe use different names.

Reply:
Welcome to Python :) this means that you're binding the local result_dir variable to the result_dir argument run() takes.